src/transport/SecureSessionMgr.cpp

```diff
@@ -384,6 +406,44 @@ void SecureSessionMgr::SecureMessageDispatch(const PacketHeader & packetHeader,
         ExitNow(err = CHIP_ERROR_KEY_NOT_FOUND_FROM_PEER);
     }
 
+    if (packetHeader.GetFlags().Has(Header::FlagValues::kSecure))
+    {
+        // Verify message counter
+        if (packetHeader.GetFlags().Has(Header::FlagValues::kSecureSessionControlMessage))
+        {
+            // TODO: control message counter is not implemented yet
+        }
+        else
+        {
+            // TODO: "initial Session Establishment bootstrap" is under specified, use message counter sync protocol for both group
+            // and unicast messages
```

Comment:
I don't understand this part.  When we do the CASE or PASE handshake we should pre-initialize the counters for the resulting secure session.  We should absolutely not be doing counter sync for non-group-key messages.

Reply:
Yes, this is the case. But currently, CASE handshake doesn't pre-initialize the counters, so this is the workaround. After we have fully implemented the CASE handshake, then the counter should be initialized by CASE. Then `IsSynchronized` should return true here.

So I'll remove the TODO comment.

Reply:
How hard would it be to pre-initialize the counters in CASE?  That seems like it would be pretty simple, and much better than doing the extra syncing....